src/contracts/multichain/OperatorTableUpdater.sol

```diff
@@ -0,0 +1,289 @@
+    ) OperatorTableUpdaterStorage(_bn254CertificateVerifier, _ecdsaCertificateVerifier) SemVerMixin(_version) {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initializes the OperatorTableUpdater
+     * @param owner The owner of the OperatorTableUpdater
+     * @param _globalRootConfirmerSet The operatorSet which certifies against global roots
+     * @param _globalRootConfirmationThreshold The threshold, in bps, for a global root to be signed off on and updated
+     * @param referenceTimestamp The reference timestamp for the global root confirmer set
+     * @param globalRootConfirmerSetInfo The operatorSetInfo for the global root confirmer set
+     * @param globalRootConfirmerSetConfig The operatorSetConfig for the global root confirmer set
+     * @dev We also update the operator table for the global root confirmer set, to begin signing off on global roots
+     */
+    function initialize(
+        address owner,
+        OperatorSet calldata _globalRootConfirmerSet,
+        uint16 _globalRootConfirmationThreshold,
+        uint32 referenceTimestamp,
+        BN254OperatorSetInfo calldata globalRootConfirmerSetInfo,
+        OperatorSetConfig calldata globalRootConfirmerSetConfig
+    ) external initializer {
+        _transferOwnership(owner);
+        _setGlobalRootConfirmerSet(_globalRootConfirmerSet);
+        _setGlobalRootConfirmationThreshold(_globalRootConfirmationThreshold);
+
+        // Update the operator table for the global root confirmer set
+        bn254CertificateVerifier.updateOperatorTable(
+            _globalRootConfirmerSet, referenceTimestamp, globalRootConfirmerSetInfo, globalRootConfirmerSetConfig
+        );
+    }
+
+    /**
+     *
+     *                         ACTIONS
+     *
+     */
+
+    /// @inheritdoc IOperatorTableUpdater
+    function confirmGlobalTableRoot(
+        BN254Certificate calldata globalTableRootCert,
+        bytes32 globalTableRoot,
+        uint32 referenceTimestamp
+    ) external {
+        // Table roots can only be updated for current or past timestamps and after the latest reference timestamp
+        require(referenceTimestamp <= block.timestamp, GlobalTableRootInFuture());
+        require(referenceTimestamp > latestReferenceTimestamp, GlobalTableRootStale());
+        require(globalTableRoot == globalTableRootCert.messageHash, TableRootNotInCertificate());
+
+        // Verify certificate by using the stake proportion thresholds
+        uint16[] memory stakeProportionThresholds = new uint16[](1);
+        stakeProportionThresholds[0] = globalRootConfirmationThreshold;
+        bool isValid = bn254CertificateVerifier.verifyCertificateProportion(
+            _globalRootConfirmerSet, globalTableRootCert, stakeProportionThresholds
+        );
+
+        require(isValid, CertificateInvalid());
+
+        // Update the global table root
+        latestReferenceTimestamp = referenceTimestamp;
+        _currentGlobalTableRoot = globalTableRoot;
```

Comment:
can we remove this and do `_globalTableRoots[latestReferenceTimestamp]` from `getCurrentGlobalTableRoot` instead?